erpnext/public/js/controllers/transaction.js

```diff
@@ -903,20 +903,73 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 
 	shipping_rule: function() {
 		var me = this;
-		if(this.frm.doc.shipping_rule) {
+		if(this.frm.doc.shipping_rule) {			
+			if(localStorage) {
+				var last_shipping_rule = localStorage.getItem("last_shipping_rule");					
+				localStorage.removeItem("last_shipping_rule");
+			}			
 			return this.frm.call({
 				doc: this.frm.doc,
 				method: "apply_shipping_rule",
 				callback: function(r) {
 					if(!r.exc) {
+						if(last_shipping_rule && cur_frm.doc.shipping_rule != last_shipping_rule)	{													
```

Comment:
```suggestion
						if(last_shipping_rule && cur_frm.doc.shipping_rule != last_shipping_rule)	{													
```
since i suggested last_shipping_rule to declare as let. then this value will get undefined so look for that.